invoicer_project/invoices/views.py

```diff
@@ -0,0 +1,158 @@
+        data = {'content': invoices_json.data}
+        return JsonResponse(data, status=200, safe=False)
+
+    def create(self, request):
+        data = JSONParser().parse(request)
+        data["user"] = get_user_from_jwt(request.headers)['user_id']
+        serializer = self.serializer_class(data=data)
+        if serializer.is_valid():
+            serializer.save()
+            return JsonResponse(serializer.data, status=201)
+        return JsonResponse(serializer.errors, status=400)
+
+    def destroy(self, request):
+        user = get_user_from_jwt(request.headers)
+        invoice_ids = request.data['elementsIds']
+        invoices = Invoice.objects.filter(id__in=invoice_ids, user_id=user['user_id'])
+        invoices.delete()
+        return JsonResponse({}, status=204)
+
+    def download_data(self, request, invoice_id):
+        try:
+            user = get_user_from_jwt(request.headers)
+            invoice = self.queryset.get(id=invoice_id, user_id=user['user_id'])
+            ordered_items = OrderedItem.objects.filter(invoice=invoice_id)
+        except Invoice.DoesNotExist:
+            return JsonResponse({'error': 'Invoice not found'}, status=404)
+        serializer = InvoiceSerializer(invoice, data=request.data, partial=True)
+        items = OrderedItemSerializer(ordered_items, many=True)
+        if serializer.is_valid():
+            serializer.save()
+        buffer = BytesIO()
+        p = canvas.Canvas(buffer, pagesize=letter)
+        pdfmetrics.registerFont(TTFont('TimesNewRoman', 'static/fonts/Roboto-Regular.ttf'))
+        p.setFont('TimesNewRoman', 20)
+        p.drawString(40, 740, f"{serializer.data['name']} від {serializer.data['date_of_payment'].split('-')[2]}.{serializer.data['date_of_payment'].split('-')[1]}.{serializer.data['date_of_payment'].split('-')[0]}р.")
+        p.line(40, 730, 565, 730)
+        p.setFont('TimesNewRoman', 13)
+        p.drawString(40, 710, f"Постачальник:  {serializer.data['user_first_name']} {serializer.data['user_last_name']}")
+        p.drawString(40, 690, f"Покупець:  {serializer.data['client_first_name']} {serializer.data['client_last_name']}")
+        table_headers = ['№', 'Назва товару', 'К-сть', 'Од.', 'Ціна', 'Сума']
+        table_header_x = 40
+        table_header_y = 670
+        table_row_x = table_header_x
+        table_row_y = table_header_y - 30
+
+        id_width = 30
+        name_width = 300
+        amount_width = 50
+        unit_width = 40
+        price_width = 45
+        sum_width = 60
+
+        p.setFont('TimesNewRoman', 11)
+        for i, data in enumerate(items.data):
+            item_id = str(i + 1)
+            name = data['item_name']
+            price = str(data['price'])
+            amount = str(data['amount'])
+            unit = str(data['unit'])
+            sum_price = str(data["price"]*data["amount"])
+
+            if i == 0:
+                p.rect(table_row_x, table_row_y, id_width, 20)
```

Comment:
what is 20? it is magic number. please, don't use magic number anymore. replace it to const

Reply:
Done